### DISCUSSION TITLE
Error while compiling examples on Ubuntu 20.04 for release v20.48.0

### DISCUSSION THREAD

**Discussion by avipl** [uNetworking/uWebSockets#1674] [Q&A] [answered]
Can someone please help me with this error? I have g++ 11.4.0 and also clang 14.0.0-1ubuntu1.1

```
./build examples
--> g++ -march=native -O3 -Wpedantic -Wall -Wextra -Wsign-conversion -Wconversion -std=c++20 -Isrc -IuSockets/src -flto examples/Http3Server.cpp  uSockets/*.o -lz -o Http3Server

/usr/bin/ld: cannot find -lz: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [GNUmakefile:7: examples] Error 255

```

I'm trying to set up a server for my IoT project. This project has no beginner guide or where to start so, I've been stuck for almost a day now.

**Comment by uNetworkingAB** [comment, marked as answer]
> /usr/bin/ld: cannot find -lz: No such file or directory

You need zlib or flags https://github.com/uNetworking/uWebSockets/blob/master/misc/READMORE.md#compiling

Btw the project is not intended for beginners. That's not the target audience.

**Reply by avipl** [reply]
I was able to compile with a GCC docker image but after disabling zlib flag it compiled on my machine.

Thank you! appreciate your response.
